### PR TITLE
Feat locale refactor

### PR DESCRIPTION
**Pull request by christyjacob4**
## What does this PR do?
- [x] Refactor all the locale templates
- [x] Use the new email templates instead of passing a path to `Template()` in  

https://github.com/appwrite/appwrite/blob/16b3c5c3a46e4ad5a5220898e90575a3bd95715f/app/controllers/api/account.php#L1723

## Test Plan
Use Existing tests

## Related PRs and Issues
NA

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

YES

### CODE REVIEW

File: app/config/locale/terms.json [hunk 1]
```diff
@@ -0,0 +1,1274 @@
+[
+    {
+        "term": "settings.inspire",
+        "context": "",
```
Comment (eldadfux) [inline app/config/locale/terms.json:4]: The PR looks great, I would only think how can we make the metadata here much more useful for translator. Can we include a description? Maybe a highlighted screenshot with the relevant area in the UI once we translate the console?

Comment (christyjacob4) [reply app/config/locale/terms.json:4]: We can add information about a key in the "context" property like so 
 
Screenshot from 2021-07-23 14-21-53


We can attach images in the comments section like so 

Screenshot from 2021-07-23 14-23-14

Comment (eldadfux) [reply app/config/locale/terms.json:4]: Sounds good

File: app/controllers/api/account.php [hunk 2]
```diff
@@ -967,8 +958,8 @@
             $record = $geodb->get($log['ip']);
 
             if ($record) {
-                $output[$i]['countryCode'] = (isset($countries[$record['country']['iso_code']])) ? \strtolower($record['country']['iso_code']) : '--';
-                $output[$i]['countryName'] = (isset($countries[$record['country']['iso_code']])) ? $countries[$record['country']['iso_code']] : $locale->getText('locale.country.unknown');
+                $output[$i]['countryCode'] = $locale->getText('countries.'.strtolower($record['country']['iso_code']), '--');
```
Comment (eldadfux) [inline app/controllers/api/account.php:961]: I think here we just need to get the country code not the translated name.

File: src/Appwrite/Template/Template.php [hunk 3]
```diff
@@ -5,8 +5,57 @@
...
+     * fromHtmlString
+     *
+     * Creates a new Template() using raw HTML.
+     * 
+     * @param string $html 
+     *
+     * @return self
+     * 
+     */
+    public static function fromHtmlString(string $html): self
```
Comment (eldadfux) [inline src/Appwrite/Template/Template.php:49]:
```suggestion
    public static function fromString(string $html): self
```

HTML should be completely optional.

File: src/Appwrite/Template/Template.php [hunk 4]
```diff
@@ -5,8 +5,57 @@
...
+    {
+        if (empty($html)) {
+            throw new Exception('Empty HTML string');
```
Comment (eldadfux) [inline src/Appwrite/Template/Template.php:52]:
```suggestion
            throw new Exception('Empty string');
```

File: src/Appwrite/Template/Template.php [hunk 5]
```diff
@@ -125,4 +176,36 @@ public static function fromCamelCaseToDash($input): string
...
+
+    /**
+     * setHtml
+     *
+     * Set the Raw html used by this template  
+     *
+     * @return string
+     * 
+     */
+    public function setHtml(string $html): self
```
Comment (eldadfux) [inline src/Appwrite/Template/Template.php:188]: Don't see a reason to have a dedicated method for this. We can just set the content directly in the  `fromString` method

File: src/Appwrite/Template/Template.php [hunk 6]
```diff
@@ -125,4 +176,36 @@ public static function fromCamelCaseToDash($input): string
...
+
+     /**
+     * getHtml
+     *
+     * Get the Raw html that this template holds.  
+     *
+     * @return string
+     * 
+     */
+    public function getHtml(): string
```
Comment (eldadfux) [inline src/Appwrite/Template/Template.php:202]: Again as the above comment, not sure why we need dedicated setters as getters, seems like there is no much code repetitiveness this is saving us. Probably better avoid coupling with HTML, this template can be used with any view format.

File: src/Appwrite/Template/Template.php [hunk 7]
```diff
@@ -5,8 +5,57 @@
 use Exception;
 use Utopia\View;
 
+use function PHPUnit\Framework\isReadable;
```
Comment (eldadfux) [inline src/Appwrite/Template/Template.php:8]:
```suggestion
```

Leftovers

File: composer.json [hunk 8]
```diff
@@ -45,7 +45,7 @@
         "utopia-php/cache": "0.2.*",
         "utopia-php/cli": "0.11.*",
         "utopia-php/config": "0.2.*",
-        "utopia-php/locale": "0.3.*",
+        "utopia-php/locale": "dev-feat-accept-file-path",
```
Comment (eldadfux) [inline composer.json:48]: We can set it to v0.4.0

File: app/controllers/api/users.php [hunk 9]
```diff
@@ -322,8 +318,8 @@
             $record = $geodb->get($log['ip']);
 
             if ($record) {
-                $output[$i]['countryCode'] = (isset($countries[$record['country']['iso_code']])) ? \strtolower($record['country']['iso_code']) : '--';
-                $output[$i]['countryName'] = (isset($countries[$record['country']['iso_code']])) ? $countries[$record['country']['iso_code']] : $locale->getText('locale.country.unknown');
+                $output[$i]['countryCode'] = $locale->getText('countries.'.strtolower($record['country']['iso_code']), '--');
```
Comment (eldadfux) [inline app/controllers/api/users.php:321]: Again, this portion of code just needs the country code.

File: app/controllers/api/locale.php [hunk 10]
```diff
@@ -42,7 +40,8 @@
 
         if ($record) {
             $output['countryCode'] = $record['country']['iso_code'];
-            $output['country'] = (isset($countries[$record['country']['iso_code']])) ? $countries[$record['country']['iso_code']] : $locale->getText('locale.country.unknown');
+            $output['country'] = $locale->getText('countries.'.strtolower($record['country']['iso_code']), $locale->getText('locale.country.unknown'));
```
Comment (eldadfux) [inline app/controllers/api/locale.php:43]: Why did we gave up the isset method check? We better be safe than sorry :)

Comment (eldadfux) [reply app/controllers/api/locale.php:43]: This has multiple instances across the diff, I'll avoid commenting on each one :)

File: tests/e2e/Services/Locale/LocaleBase.php [hunk 11]
```diff
@@ -159,8 +159,8 @@ public function testGetContinents():array
         $this->assertEquals($response['headers']['status-code'], 200);
         $this->assertEquals(7, $response['body']['sum']);
         $this->assertIsArray($response['body']['continents']);
-        $this->assertEquals($response['body']['continents'][0]['code'], 'NA');
-        $this->assertEquals($response['body']['continents'][0]['name'], 'América del Norte');
+        $this->assertEquals($response['body']['continents'][4]['code'], 'NA');
```
Comment (eldadfux) [inline tests/e2e/Services/Locale/LocaleBase.php:162]: We should make sure we're sorting the list using the locale ABC

**Review by eldadfux — CHANGES_REQUESTED** [review]

File: app/controllers/api/account.php (hunk 2 above)
Comment (christyjacob4) [reply app/controllers/api/account.php:961]: Done :+1:

File: src/Appwrite/Template/Template.php (hunk 3 above)
Comment (christyjacob4) [reply src/Appwrite/Template/Template.php:49]: Done :+1:

File: src/Appwrite/Template/Template.php (hunk 4 above)
Comment (christyjacob4) [reply src/Appwrite/Template/Template.php:52]: done :+1:

File: src/Appwrite/Template/Template.php (hunk 5 above)
Comment (christyjacob4) [reply src/Appwrite/Template/Template.php:188]: Yes but I've defined `fromString` to be a static method. `setHtml()` would act on an instance. Let me know if this is not required.

File: src/Appwrite/Template/Template.php (hunk 7 above)
Comment (christyjacob4) [reply src/Appwrite/Template/Template.php:8]: done :+1:

File: composer.json (hunk 8 above)
Comment (christyjacob4) [reply composer.json:48]: Done :+1:

File: app/controllers/api/users.php (hunk 9 above)
Comment (christyjacob4) [reply app/controllers/api/users.php:321]: Screenshot from 2021-07-26 10-47-03


I dont think so... 
`$countries = $locale->getText('countries'); `

which returns 
```php
[
   "IN" => "India"
]
```

Comment (christyjacob4) [reply app/controllers/api/users.php:321]: Done :+1:

File: app/controllers/api/locale.php (hunk 10 above)
Comment (christyjacob4) [reply app/controllers/api/locale.php:43]: `$locale->getText()` takes a default param. So I thought we can use that instead :+1:

File: tests/e2e/Services/Locale/LocaleBase.php (hunk 11 above)
Comment (christyjacob4) [reply tests/e2e/Services/Locale/LocaleBase.php]: Done :+1:

**Pull request merged**